src/routes/provider-edit-route/provider-edit-route.test.js

```diff
@@ -0,0 +1,269 @@
+  isLoading: false,
+  proxy: {
+    id: 'proxy-id',
+    inherited: false,
+  },
+  providerToken: {
+    value: 'token-value',
+  },
+  update: {
+    errors: [],
+    isPending: false,
+    isRejected: false,
+    isResolved: false,
+  },
+  request: {
+    errors: [],
+    isPending: false,
+    isRejected: false,
+    isResolved: true,
+  },
+  destroy: {
+    errors: [],
+    isRejected: false,
+  },
+  visibilityData: {
+    isHidden: false,
+  },
+};
+
+const proxyTypes = {
+  request: {
+    isResolved: true,
+  },
+  resolver: {
+    state: {
+      proxyTypes: {
+        records: {
+          'proxy-id': {
+            id: 'proxy-id',
+            attributes: {
+              name: 'Some Proxy',
+            },
+          },
+          'proxy-id-2': {
+            id: 'proxy-id-2',
+            attributes: {
+              name: 'Some Other Proxy',
+            },
+          },
+        },
+      },
+    },
+  },
+};
+
+const rootProxy = {
+  data: {
+    attributes: {
+      proxyTypeId: 'ezproxy',
+    },
+  },
+  request: {
+    isResolved: true,
+  },
+};
+
+const mockGetProvider = jest.fn();
+const mockGetProxyTypes = jest.fn();
+const mockGetRootProxy = jest.fn();
+const mockRemoveUpdateRequests = jest.fn();
+const mockUpdateProvider = jest.fn();
+
+const getProviderEditRoute = (props = {}) => (
+  <MemoryRouter>
+    <Harness>
+      <ProviderEditRoute
+        getProvider={mockGetProvider}
+        getProxyTypes={mockGetProxyTypes}
+        getRootProxy={mockGetRootProxy}
+        history={history}
+        location={location}
+        match={match}
+        model={model}
+        proxyTypes={proxyTypes}
+        removeUpdateRequests={mockRemoveUpdateRequests}
+        rootProxy={rootProxy}
+        updateProvider={mockUpdateProvider}
+        {...props}
+      />
+        Page content
+    </Harness>
+  </MemoryRouter>
+);
+
+const renderProviderEditRoute = (props) => render(getProviderEditRoute(props));
+describe('Given ProviderEditRoute', () => {
+  beforeEach(() => {
+    mockGetProxyTypes.mockClear();
+    mockGetRootProxy.mockClear();
+    mockRemoveUpdateRequests.mockClear();
+    mockUpdateProvider.mockClear();
+    history.replace.mockClear();
+  });
+
+  afterEach(cleanup);
+
+  it('should render page', () => {
+    const { getByText } = renderProviderEditRoute({});
+
+    expect(getByText('Page content')).toBeDefined();
+  });
+
+  it('should handle getProvider with providerId', async () => {
+    await act(async () => {
+      await renderProviderEditRoute();
+    });
+
+    expect(mockGetProvider).toHaveBeenCalledWith('provider-id');
+  });
+
+  it('should handle getProvider', async () => {
+    await act(async () => {
+      await renderProviderEditRoute({
+        props: {
+          getProvider: mockGetProvider,
+          match: {
+            ...match,
+            params: { providerId: 'other-provider-id' },
+          },
+        },
+      });
+    });
+
+    expect(mockGetProvider).toHaveBeenCalled();
```

Comment:
Please also test that this prop is called with correct arguments.
Sometimes it's very important to test this, for example when requesting data from backend we need to make sure that the right data is requested, or when a child calls a prop with some data to pass to parent - we also need to make sure that data is correct, or when redirecting to a new page
If you check `expect(mockGetProvider).toHaveBeenCalledWith('other-provider-id');` it'll fail because of how you've passed props to component - see my next comment about this